ttnn/cpp/ttnn/operations/conv/conv_transpose2d/conv_transpose2d.cpp

```diff
@@ -0,0 +1,296 @@
+        auto out_channels  = in_w_shape[1];
+        auto kernel_height = in_w_shape[2];
+        auto kernel_width  = in_w_shape[3];
+        ttnn::SimpleShape output_shape{out_channels, in_channels, kernel_height, kernel_width};
+        auto output_buffer = owned_buffer::create<T>(output_shape.volume());
+
+        for(auto in_kernel_height_index = 0; in_kernel_height_index < kernel_height; in_kernel_height_index++)  {
+            auto out_buffer_kh_index = kernel_height - in_kernel_height_index - 1;
+            for(auto in_kernel_width_index = 0; in_kernel_width_index < kernel_width; in_kernel_width_index++)  {
+                auto out_buffer_kw_index = kernel_width - in_kernel_width_index - 1;
+                for(auto in_channels_index = 0; in_channels_index < in_channels; in_channels_index++)   {
+                    for(auto out_channels_index = 0; out_channels_index < out_channels; out_channels_index++)   {
+                        auto in_idx = in_channels_index * kernel_height * kernel_width * out_channels + out_channels_index * kernel_height * kernel_width + in_kernel_height_index * kernel_width + in_kernel_width_index;
+                        auto out_idx = out_channels_index * in_channels * kernel_height * kernel_width + in_channels_index * kernel_height * kernel_width + out_buffer_kh_index * kernel_width + out_buffer_kw_index;
+                        output_buffer[out_idx] = input_buffer[in_idx];
+                    }
+                }
+            }
+        }
+        return Tensor(std::move(OwnedStorage{std::move(output_buffer)}), output_shape, dtype, Layout::ROW_MAJOR);
+    };
+    auto convert_tensor = [&compute](const auto& conv_weight_tensor) {
+        return std::visit(
+            [&compute](auto&& storage) -> Tensor {
+                using StorageType = std::decay_t<decltype(storage)>;
+                if constexpr (std::is_same_v<StorageType, OwnedStorage>) {
+                    return compute(owned_buffer::get_as<T>(storage.buffer));
+                } else if constexpr (std::is_same_v<StorageType, BorrowedStorage>) {
+                    return compute(borrowed_buffer::get_as<T>(storage.buffer));
+                } else {
+                    TT_THROW("Unsupported storage type");
+                }
+            },
+            conv_weight_tensor.get_storage());
+    };
+    return is_multi_device_tensor(conv_weight_tensor) ? transform(conv_weight_tensor, convert_tensor) : convert_tensor(conv_weight_tensor);
+}
+
+
+Tensor transform_weights_for_conv_transpose2d(const Tensor& conv_weight_tensor) {
+            switch (conv_weight_tensor.get_dtype()) {
+                case DataType::BFLOAT16:
+                    return _transform_weights_for_conv_transpose2d<::bfloat16>(conv_weight_tensor);
+                case DataType::FLOAT32:
+                    return _transform_weights_for_conv_transpose2d<float>(conv_weight_tensor);
+                case DataType::UINT32:
```

Comment:
do we support integer weights?

Reply:
I don't think the op supports it. But the existing weights preparation code [does](https://github.com/tenstorrent/tt-metal/blob/4e1fef96b1fc5543455911fef6fcf7664f04d9b3/ttnn/cpp/ttnn/tensor/tensor_utils.cpp#L204).